### PR TITLE
chore: add modules and config for config_split

### PR DESCRIPTION
**Pull request by lazysoundsystem**
Refs: OPS-7850

Inline with changes to numbers-site https://github.com/UN-OCHA/numbers-site/pull/84 adding the 'standard' config for config_split. This doesn't include `stage_file_proxy` and does include `dev_mode` so we can improve the standardization, but this is a start.

### CODE REVIEW

**Comment by lazysoundsystem** [comment]
This was painful to debug - I was getting an error in the tests, but only within the test environment (i.e. my local, set up with the script here, could run `composer install` happily) and once I hit the problem I was getting it on the main branch too:
```
$ docker exec -it -w /srv/www starterkit-test-site composer install

Gathering patches from patch file.

Installing dependencies from lock file (including require-dev)

Verifying lock file contents can be installed on current platform.

Nothing to install, update or remove

Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.

Package symfony/debug is abandoned, you should avoid using it. Use symfony/error-handler instead.

Package webmozart/path-util is abandoned, you should avoid using it. Use symfony/filesystem instead.

Generating autoload files

Fatal error: Uncaught TypeError: count(): Argument #1 ($value) must be of type Countable|array, Closure given in /srv/www/vendor/composer/composer/src/Composer/Autoload/AutoloadGenerator.php:1202

Stack trace:

#0 /srv/www/vendor/composer/composer/src/Composer/Autoload/AutoloadGenerator.php(1202): count(Object(Closure))
```
from: https://app.travis-ci.com/github/UN-OCHA/drupal-starterkit/builds/259085529

**Comment by attiks** [comment]
Problem is composer 2.5

**Comment by orakili** [comment]
@lazysoundsystem As pointed by Attiks, the issue was with the `composer self-update` in the `docker/Dockerfile`. I removed as it's unnecessary. It's a leftover of an old issue with composer: https://github.com/drupal-composer/drupal-project/issues/282.

I also had to explicitly  install `config_filter` to be able to import the config so I added it the `dev` dependencies in the composer like it's done on some other sites (ex: RW9).

**dev_mode module**

TL;DR; this module is a bad idea and should not be in the starterkit.

I also removed the `dev_mode` from the config_split settings file to prevent it from being enabled by default because it makes a mess as it tries to override the `settings.php` and change permissions.

In addition there is a setting in the `settings.local.php` to prevent its configuration (i.e. it being enabled by default when importing the config) from being saved so that it doesn't end up again in the conflig_split or core.extension settings.

When enabled, this `dev_mode` module modifies some settings in the database that are at risk of being exported and saved by mistake. For example  the `system.performance` and `system.logging` settings.

I think this module is a really bad idea or at least badly executed. For example, if you want to only partially disable some elements (ex: test js/css aggregation while still having the page cache being disabled) then it becomes messy because of the order of the `settings.php` files (settings.php -> settings.local.php -> settings.dev_mode.php).

There are not so many differences between what we already provide in the `settings.local.php` and what is in the `settings.dev_mode.php` file. Basically the render, page etc. cache bins are set to `cache.backend.null` and that's about it.

So basically either one runs `drush en dev_mode` and risks shooting themselves in the foot when exporting config if not careful or one edits the `settings.local.php` and uncomment/comment a few lines.

In any case, I'm merging this PR because it works and will port it to the D10 version of the startkit. But I think there should be a follow-up PR to remove the `dev_mode` module.

**Review by orakili — APPROVED** [review]
